src/common/allocator.h

```diff
@@ -4,7 +4,11 @@
 #ifndef CEPH_COMMON_ALLOCATOR_H
 #define CEPH_COMMON_ALLOCATOR_H
 
+#if defined(__linux__)
```

Comment:
@dillaman @wjwithagen  since we will add a guard here anyway, can we put something like
```c++
#include "acconfig.h"

#ifdef LIBTCMALLOC_MISSING_ALIGNED_ALLOC
#include <malloc.h>
#include <new>
#endif
#include <memory>
```

Reply:
Yup, that should also work.